test/core/runtime/stack_limiter_test.cpp

```diff
@@ -250,3 +273,47 @@ INSTANTIATE_TEST_SUITE_P(SuiteFromSubstrate,
                                          "simple",
                                          "start",
                                          "table"));
+
+auto wat_memory_import = R"(
+  (module
+    (import "env" "mem" (memory (;0;) 100)))
+)";
+auto wat_memory_export = R"(
+  (module
+    (memory (;0;) 100)
+    (export "mem" (memory 0)))
+)";
+auto memory_limit_static = std::make_pair(HeapAllocStrategyStatic{100}, R"(
+  (module
+    (memory (;0;) 200 200)
+    (export "mem" (memory 0)))
+)");
+
+TEST(WasmInstrumentTest, memory_import) {
```

Comment:
Please comment the tests with given/when/then notation or at least add a brief comment for each case (the first option is more anticipated).